bin/assert-content-auth.sh

```
@@ -0,0 +1,158 @@
+    start_nginx
+
+    # Prepare curl command
+    local curl_cmd="curl --silent --output /dev/null --write-out %{http_code}|%{content_type}|%{redirect_url}"
+
+    # Add auth token if provided
+    if [ -n "$auth_token" ]; then
+        curl_cmd="$curl_cmd --oauth2-bearer $auth_token"
+    fi
+
+    # Make the request
+    local response=$($curl_cmd "http://localhost:${PORT}${path}")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)